source/server/server.cc

```diff
@@ -412,9 +412,32 @@ void InstanceImpl::initialize(const Options& options,
     server_compilation_settings_stats_->fips_mode_.set(1);
   }
 
-  bootstrap_.mutable_node()->set_hidden_envoy_deprecated_build_version(VersionInfo::version());
-  bootstrap_.mutable_node()->set_user_agent_name("envoy");
-  *bootstrap_.mutable_node()->mutable_user_agent_build_version() = VersionInfo::buildVersion();
+  // Check to see if bootstrap has set user_agent_name. If yes, use that.
+  // If not, use "envoy"
+  if (bootstrap_.node().user_agent_name().empty()) {
+    bootstrap_.mutable_node()->set_user_agent_name("envoy");
+  } else {
+    bootstrap_.mutable_node()->set_user_agent_name(bootstrap_.node().user_agent_name());
```

Comment:
Why do you need the `else` statement? It looks noop.

Reply:
Looks like you're right for 2 of the 3.  You'll have to bear with me for a little; first time really working with this codebase and all he protos and generated bits.  I was a bit confused on the `node` vs `mutable_node` and which 

Looks like `user_agent_name` and `user_agent_build_version` both only need to be set IF the user hasn't set anything.  

However, the `user_agent_version` seems to be different due to the user of the `set_hidden_deprecated_build_version` bit.  I don't quite know why, but regardless of if `bootstrap_.node().user_agent_version()` has been set, it seems we still need to call `bootstrap_.mutable_node()->set_hidden_envoy_deprecated_build_version()` to get it to pop out in the `/config_dump`.  

Related, though not necessarily part of this issue: it is odd to me that the output node information from /config_dump doesn't match the input for this field.  It actually dumps out `hidden_envoy_deprecated_build_version` rather than `user_agent_version`

```json
    "node": {
     "id": "bootstrap_id",
     "cluster": "bootstrap_cluster",
     "hidden_envoy_deprecated_build_version": "test",
     "user_agent_name": "test-ci-user-agent",
     "user_agent_build_version": {
      "version": {
       "major_number": 1,
       "minor_number": 19
      },

```